docs/DeveloperGuide.md

```diff
@@ -427,15 +446,106 @@ object.
+
+Given below is a series of steps to show how a `science` operation behaves during its execution.
+
+1. The user enters the `science` command string.
+
+2. This calls the `execute` method of the `LogicManager` class with the user input passed in as a string.
+
+3. `Logic.execute()` then calls the `parseCommand`  method of the `GradPadParser` class to parse the string input.
+
+4. `GradPadParser.parseCommand()` identifies the command as a science command and thus creates a `ScienceCommand`
+object.
+
+5. This command object is then passed back to the `LogicManager` in step 2.
+
+6. `LogicManager` executes the newly created `ScienceCommand`, which will contain a list of `scienceModules` hard coded in.
+
+7. Then, the `ScienceCommand.execute()` calls its own method, `setScienceModules` which sets the `scienceModules` with 
+the list of available modules.
 
-8. It filters the list of required modules by removing those that are present in `modules`.
+8. Finally, a `CommandResult` is created with the `scienceModules` to display the modules.
 
-9. Finally, a `CommandResult` is created to show the filtered list of remaining required modules.
+The following sequence diagram illustrates how the `science` command is executed.
+
+![ScienceDiagram](images/ScienceSequenceDiagram.png)
+
+### Check all available General Education Modules
+
+The `gem` command allows users to view all available General Education (GE) modules available in NUS.
+
+When the command is executed, a list of all available GE modules will be displayed on the `Command Line Display`.
+
+This is achieved with the `GemCommand` and `GemCommandStorage` class. The `GemCommandStorage` class
+handles the extracting and parsing of JSON module data while the `GemCommand` handles the logic of displaying the
+modules. This command is separated from `required` to avoid cluttering up the `Command Line Display` due to the hefty
+amount of GE modules displayed.
+
+As with all operations in GradPad, the `GemCommand` class handles the execution of `gem` operations.
+
+Given below is a series of steps to show how a `gem` operation behaves during its execution.
+
+1. The user enters the `gem` command string.
+
+2. This calls the `execute` method of the `LogicManager` class with the user input passed in as a string.
+
+3. `Logic.execute()` then calls the `parseCommand`  method of the `GradPadParser` class to parse the string input.
+
+4. `GradPadParser.parseCommand()` identifies the command as a gem command and thus creates a `GemCommand`
+object.
+
+5. This command object is then passed back to the `LogicManager` in step 2.
+
+6. `LogicManager` executes the newly created `GemCommand`, which will contain 2 `GemCommandStorage` attributes,
+`sem1Storage` and `sem2Storage`.
+
+7. `GemCommand.execute()` then calls its own method `setSem1Storage` and `setSem2Storage` to create 2 `GemCommandStorage`
+objects.
```

Comment:
I was abit confused when step 6 says `GemCommand` already contains both storages, but step 7 says they're only created then. Maybe we don't have to mention the attributes? I think the creation of both objects is quite clear on its own

Reply:
ya my bad HAHAH, I was trying to say it will eventually have those 2 storages. phrasing is bad on my part. I changed it to uninitialised storage attributes w/o the names, to make it clearer.